static/js/message_scroll.js

```diff
@@ -166,7 +166,7 @@ exports.initialize = function () {
 
     // Scroll handler that marks messages as read when you scroll past them.
     $(document).on("message_selected.zulip", (event) => {
-        if (event.id === -1) {
+        if (event.id === -1 || event.id === event.previously_selected) {
```

Comment:
I wonder if a good follow up fix here would be to rename `event.id` and `event.previously_selected`.  It's not obvious that either is a message id, and when you look at this code out of context, it seems funny to compare those two things.